config/v1/types_infrastructure.go

```diff
@@ -528,18 +528,21 @@ type AWSPlatformStatus struct {
 
 // AWSResourceTag is a tag to apply to AWS resources created for the cluster.
 type AWSResourceTag struct {
-	// key is the key of the tag
+	// key sets the key of the AWS resource tag key-value pair. Key is required when defining an AWS resource tag.
+	// Length Constraints: Minimum length of 1. Maximum length of 128.
+	// The string can contain only the set of Unicode letters, digits, white-space, '_', '.', '/', '=', '+', '-', ':', '@' and cannot be prefixed with 'aws:'.
 	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:validation:MaxLength=128
-	// +kubebuilder:validation:Pattern=`^[0-9A-Za-z_.:/=+-@]+$`
+	// +kubebuilder:validation:XValidation:rule=`self.matches('^(?!aws:).+[0-9A-Za-z_.:/=+-@ ]+$')`,message="invalid AWS resource tag key. The string can contain only the set of Unicode letters, digits, white-space, '_', '.', '/', '=', '+', '-', ':', '@' and cannot be prefixed with 'aws:'"
```

Comment:
Additionally, could you add tests for the validations to make sure:
- Values we expect to be valid are actually valid
- Values we expect to be invalid are not valid

Reply:
https://github.com/openshift/api/tree/master/tests does not tell you how to actually execute the tests

Reply:
ah I see that there is a nested hack dir: https://github.com/openshift/api/blob/master/tests/hack/test.sh but that is panicking because I don't have the envtest dependencies

Reply:
ah, so I just needed to use `make` and that runs all the tests. that pulled in my new tests which failed (as expected), so that's good. it took 16m, so unless there is some caching that takes place on subsequent runs I need to figure out how to run individual tests

Reply:
I'm newer to the API review process for OpenShift so I don't know off the top of my head how to run individual tests. Maybe @JoelSpeed would have something handy?

Reply:
So I normally use the Makefile in each API group, eg https://github.com/openshift/api/blob/master/config/v1/Makefile

With `make -C config/v1 test`, or you can do what it is doing yourself if you want

```
GINKGO_EXTRA_ARGS="--focus=something that is in my test case" make -C tests test
```

TBH for dev I normally just adjust the focus within the makefile myself